master/internal/rm/agentrm/agent.go

```diff
@@ -129,10 +132,46 @@ func (a *agent) receive(ctx *actor.Context, msg interface{}) error {
 		a.slots, _ = ctx.ActorOf("slots", &slots{})
 	case model.AgentSummary:
 		ctx.Respond(a.summarize(ctx))
-	case ws.WebSocketConnected:
-		check.Panic(check.True(a.socket == nil, "websocket already connected"))
-		socket, ok := msg.Accept(ctx, aproto.MasterMessage{}, true)
-		check.Panic(check.True(ok, "failed to accept websocket connection"))
+	case actorapi.WebSocketRequest:
+		if a.socket != nil {
+			err := errors.New("websocket already connected")
+			ctx.Log().WithError(err).Error("socket not nil when WebSocketRequest received")
```

Comment:
is there a reason to handle errors more than once, throughout this case?

Reply:
If an error is returned to shut the actor down, nothing gets logged. So in this case I logged the errors so it wouldn't be a silent failure and so that it would replicate the original behavior as closely as possible without panics. Happy to change this if you don't think that's best.